### PR TITLE
Expose Policy.Func to public

### PR DESCRIPTION
**Pull request by MartinForReal**
close: #21057 

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by MartinForReal** [comment]
https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md is a dead link. Which change log should I update?

**Comment by jhendrixMSFT** [comment]
Each module has its own changelog.  The one for `azcore` is [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/CHANGELOG.md).

**Comment by MartinForReal** [comment]
change log has been updated.

File: sdk/azcore/policy/policy.go [hunk 1]
```diff
@@ -19,6 +19,9 @@ import (
 // Request and react to the received Response.
 type Policy = exported.Policy
 
+// PolicyFunc is a type that implements the Policy interface. Use this type when implementing a stateless policy as a first-class function.
+type PolicyFunc = exported.PolicyFunc
```
Comment (jhendrixMSFT) [inline sdk/azcore/policy/policy.go:23]: This type name stutters, it should probably be `Func`, making the fully qualified type name `policy.Func`.

Comment (MartinForReal) [reply sdk/azcore/policy/policy.go:23]: I think it will be ambiguous because we don't know from its name what Func does.

Comment (jhendrixMSFT) [reply sdk/azcore/policy/policy.go:23]: The fully qualified name will be `policy.Func`.  The name will never be just `Func` unless it's referenced from within the same package.  Compared to `policy.PolicyFunc` which stuttered (i.e. the word "policy" is repeated).

Comment (MartinForReal) [reply sdk/azcore/policy/policy.go]: updated.

**Comment by jhendrixMSFT** [comment]
@JeffreyRichter any concerns about exporting this?

**Comment by jhendrixMSFT** [comment]
Customers would use it the exact same way we do internally; to create a stateless pipeline policy from a function (example [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/runtime/pipeline.go#L67)).

Customers can achieve the same results by defining their own `pipeline.Policy` type.

**Comment by JeffreyRichter** [comment]
> Customers can achieve the same results by defining their own `pipeline.Policy` type.
Exactly, so I don't see a reason for us to public expose this rarely used and completely optional thing.

**Pull request closed without merging**